### PR TITLE
[feature](Nereids) replace order by keys by child output if possible

### PR DESCRIPTION
**Pull request by morrySnow**
# Proposed changes

To support query like that:
```sql
SELECT c1 + 1 as a, sum(c2) FROM t GROUP BY c1 + 1 ORDER BY c1 + 1
```

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.21 seconds
 load time: 440 seconds
 storage size: 17177492982 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221109081800_clickbench_pr_42588.html

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ReplaceExpressionByChildOutput.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                                    .forEach(p -> sMap.put(p.child(), p.toSlot()));
+                            return replaceSortExpression(sort, sMap);
+                        })
+                ))
+                .build();
+    }
+
+    private LogicalPlan replaceSortExpression(LogicalSort<? extends LogicalPlan> sort, Map<Expression, Slot> sMap) {
+        List<OrderKey> orderKeys = sort.getOrderKeys();
+        AtomicBoolean changed = new AtomicBoolean(false);
```
Comment (adonis0147) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ReplaceExpressionByChildOutput.java:85]: Why use `AtomicBoolean`? Does some race conditions exist?

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ReplaceExpressionByChildOutput.java:85]: lambda in stream api cannot use normal local variable

**Review by adonis0147 — APPROVED** [review]
LGTM

**Pull request merged**